language/tools/read-write-set/types/src/lib.rs

```diff
@@ -172,6 +212,31 @@ impl ReadWriteSet {
         self.iter_paths_impl(f)
     }
 
+    pub fn get_keys(&self, is_write: bool) -> Option<Vec<ResourceKey>> {
```

Comment:
It might be slightly cleaner to expose `get_keys_read` and `get_keys_written` + use the bool variant as an internal helper function. I think public API's that take a bool tend to be less user-friendly + more-error prone.